src/dotnet-svcutil/lib/src/TypeLoader.cs

```diff
@@ -32,20 +35,26 @@ static public Assembly LoadAssembly(string path)
             }
         }
 
-        static public Type[] LoadTypes(Assembly assembly, Verbosity verbosity)
+        static public Type[] LoadTypes(Assembly assembly, Verbosity verbosity, ILogger logger, OperationalContext? context)
         {
-            Type[] types;
+            List<Type> listType = new List<Type>();
 
             try
             {
-                types = assembly.GetTypes();
+                listType.AddRange(assembly.GetTypes());
             }
             catch (ReflectionTypeLoadException rtle)
             {
                 string warning;
-                types = Array.FindAll<Type>(rtle.Types, delegate (Type t) { return t != null; });
+                listType.AddRange(Array.FindAll<Type>(rtle.Types, delegate (Type t)
+                {
+                    return t != null;
+                }));
+
+                //type.Module or type.Assembly could throw if multiple assembly with same name get referenced but only one version get restored.                
+                listType = listType.Except(GetUnAvailableTypes(listType)).ToList();
 
-                if (verbosity > Verbosity.Normal)
+                if (verbosity > Verbosity.Normal || context == OperationalContext.Infrastructure)
```

Comment:
Why is this needed?

Reply:
That means we generated proxy code from visualstudio. so if  created from visualstudio, we need to record all warnings